src/store/validation/index.ts

```diff
@@ -0,0 +1,165 @@
+  validationStatuses: {} as { [key:string]: IValidationStatus },
+})
+
+export const getters: GetterTree<IValidationState, IRootState> = {
+  /**
+   * Human readable errors, keys are index like `flow/flowId/.path/.to/.prop`.
+   * Note that indexedErrors has more elements than validationStatuses.
+   */
+  flattenErrorMessages(state): IIndexedString {
+    let accumulator: IIndexedString = {}
+    forIn(state.validationStatuses, function (validationStatus: IValidationStatus, index: string) {
+      flatValidationStatuses({
+        keyPrefix: index,
+        errors: validationStatus.ajvErrors,
+        accumulator
+      })
+    })
+    return accumulator
+  }
+}
+
+export const mutations: MutationTree<IValidationState> = {
+
+}
+
+export const actions: ActionTree<IValidationState, IRootState> = {
+  async validate_block({ state, commit }, { block } : { block: IBlock }): Promise<IValidationStatus> {
+    const { uuid: blockId, type: blockType } = block
+    const blockTypeWithoutNameSpace = blockType.split('.')[blockType.split('.').length - 1]
+    const validate = getOrCreateBlockValidatorFor(blockTypeWithoutNameSpace)
+    const index = `block/${blockId}/`
+
+    Vue.set(state.validationStatuses, index, {
+      isValid: validate(block),
+      ajvErrors: validate.errors,
+    })
+
+    debugValidationStatus(state.validationStatuses[index], `validation status for ${index}`)
+    return state.validationStatuses[index]
+  },
+
+  async validate_flow({ state, commit }, { flow } : { flow: IFlow }): Promise<IValidationStatus> {
+    const validate = getOrCreateFlowValidator()
+    const index = `flow/${flow.uuid}/`
+    Vue.set(state.validationStatuses, index, {
+      isValid: validate(flow),
+      ajvErrors: validate.errors,
+    })
+
+    debugValidationStatus(state.validationStatuses[index], `flow validation status`)
+    return state.validationStatuses[index]
+  }
+}
+
+export const store: Module<IValidationState, IRootState> = {
+  namespaced: true,
+  state: stateFactory,
+  getters,
+  mutations,
+  actions,
+}
+
+export default store
+
+function getOrCreateBlockValidatorFor(blockType: string): ValidateFunction {
+  if (isEmpty(validators) || !validators.has(blockType)) {
+    const blockJsonSchemaFile = `I${blockType}Block.json`
+    // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+    validators.set(blockType, createDefaultJsonSchemaValidatorFactoryFor(require(`../../../_tmp/${blockJsonSchemaFile}`)))
+  }
+  return validators.get(blockType)!
+}
+
+function getOrCreateFlowValidator(): ValidateFunction {
+  const validationType = 'flow'
+  if (isEmpty(validators) || !validators.has(validationType)) {
+    // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+    validators.set(validationType, createDefaultJsonSchemaValidatorFactoryFor(require('../../../_tmp/flowSpecJsonSchema.json'), '#/definitions/IFlow'))
```

Comment:
@safydy Do we have a JIRA for the phase that will account for pulling these in from @floip/flow-runner repo? Did I already ask this? 😹

Reply:
My plan is to merge vmo-3702 (Flow-runner side), then edit this part before sending to QA